cmd/ctr/commands/run/run_unix.go

```diff
@@ -137,8 +141,12 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 				}
 				opts = append(opts,
 					oci.WithUserNamespace([]specs.LinuxIDMapping{uidMap}, []specs.LinuxIDMapping{gidMap}))
-				if context.Bool("read-only") {
```

Comment:
Should this remain as an `else if`?

Reply:
I was aligning the user ns path with the "main" path which has this comment a few lines below: "Even when "read-only" is set, we don't use KindView snapshot here. (#1495)"; this was never applied for the userns path when it was added, but I believe the same problem will exist (`runc` won't be able to make mount points because of a read only root fs; but *is* able to remount it `ro` once that work is completed at the `runc` level)